### PR TITLE
KZL-30 Persistent search filters across collections and reloads

### PR DESCRIPTION
**Pull request by xbill82**
## What does this PR do ?
This PR enables the Data -> Index -> Collection -> Browse view to remember search filters across collections, reloads and document edits.

#### Persistent filters across collections

1. Perform a search on a collection (quick, basic o raw) - you should see the filter specified in the URL
2. Switch to another collection
3. Switch back to the previous collection
4. The filter specified for the previous search is still defined

#### Persistent filters across reloads

1. Perform a search on a collection (quick, basic o raw)
2. The the filter is specified in the URL
3. Reload the page
4. The filter specified for the previous search is still defined

#### Persistent filters across document edits

1. Perform a search on a collection (quick, basic o raw)
2. Edit a document in the results
3. After updating the document, the results of the search are still visible

#### Persistent filters across sessions

1. Perform a search on a collection (quick, basic o raw)
2. Logout and login again
3. Browse to the previous collection
4. The filter specified for the previous search is still defined

Note that this happens even if you login as another user.

### Boyscout

A major refactoring has been performed on the search filters system.

### CODE REVIEW

File: src/components/Common/CommonList.vue [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+    },
+    paginationFrom() {
+      return parseInt(this.currentFilter.from) || 0
+    },
+    paginationSize() {
+      return parseInt(this.currentFilter.size) || 10
+    }
+  },
+  methods: {
+    isChecked(id) {
```
Comment (Aschen) [inline src/components/Common/CommonList.vue]: This shouldn't be a computed value ?

File: src/components/Common/CrudlDocument.vue [hunk 2]
```diff
@@ -117,141 +111,107 @@
...
+      )
+    },
+    confirmBulkDelete() {
+      this.isLoading = true
+      this.performDelete(this.index, this.collection, this.selectedDocuments)
+        .then(() => {
+          this.close()
+          this.refreshSearch()
+          this.isLoading = false
+          return null
```
Comment (Aschen) [inline src/components/Common/CrudlDocument.vue]: Why returning null ?
(+ others below)

File: src/components/Common/CrudlDocument.vue [hunk 3]
```diff
@@ -117,141 +111,107 @@
...
+          this.$store.commit(SET_TOAST, { text: e.message })
+        })
+    },
+    onFiltersUpdated(newFilters) {
+      this.$emit('filters-updated', newFilters)
+    },
+    dispatchToggle() {
+      this.$emit('toggle-all')
+    },
+    setBasicFilter(value) {},
```
Comment (Aschen) [inline src/components/Common/CrudlDocument.vue]: Useless method ?

File: src/components/Security/Profiles/CommonList.vue [hunk 4]
```diff
@@ -42,144 +41,155 @@
...
-        }
+    paginationFrom() {
+      return parseInt(this.$store.state.route.query.from) || 0
+    },
+    paginationSize() {
+      return parseInt(this.$store.state.route.query.size) || 10
+    }
+  },
+  methods: {
+    isChecked(id) {
```
Comment (Aschen) [inline src/components/Security/Profiles/CommonList.vue]: Should be a computed value

File: src/components/Security/Profiles/CommonList.vue [hunk 5]
```diff
@@ -42,144 +41,155 @@
...
-        return this.$store.state.route.query.searchTerm !== '' || this.basicFilter.length > 0 || this.rawFilter.length > 0
-      },
-      fetchData () {
-        let pagination = {
-          from: this.paginationFrom,
-          size: this.paginationSize
-        }
+      this.selectedDocuments.splice(index, 1)
+    },
+    hasSearchFilters() {
```
Comment (Aschen) [inline src/components/Security/Profiles/CommonList.vue]: should be a computed value

File: src/components/Common/CommonList.vue [hunk 6]
```diff
@@ -0,0 +1,236 @@
...
+
+      if (index === -1) {
+        this.selectedDocuments.push(id)
+        return
+      }
+
+      this.selectedDocuments.splice(index, 1)
+    },
+    onFiltersUpdated(newFilters) {
+      filterManager.save(newFilters, this.$router, this.index, this.collection)
```
Comment (Aschen) [inline src/components/Common/CommonList.vue]: What if `.save()` throw an exception ?

File: test/unit/specs/vuex/common/crudlDocument/mutations.spec.js [hunk 7]
```diff
@@ -2,13 +2,4 @@ import { mutations } from '../../../../../../src/vuex/modules/common/crudlDocume
...
-  describe('SET_BASIC_FILTER', () => {
-    it('should set basicFilter value', () => {
-      let state = {basicFilter: undefined}
-
-      SET_BASIC_FILTER(state, 'filter')
-      expect(state.basicFilter).to.equals('filter')
-    })
-  })
-})
+describe('collection mutations', () => {})
```
Comment (Aschen) [inline test/unit/specs/vuex/common/crudlDocument/mutations.spec.js]: Why keep empty test ?

File: src/components/Common/CommonList.vue (hunk 1 above)
Comment (xbill82) [reply src/components/Common/CommonList.vue]: No. This is a function that returns true if the document with the given id is selected. Computed values do not accept arguments.

File: src/components/Common/CrudlDocument.vue (hunk 2 above)
Comment (xbill82) [reply src/components/Common/CrudlDocument.vue]: I don't know. Even if the blame points on me, I didn't code these methods, I just formatted them via Prettier.

File: src/components/Common/CrudlDocument.vue (hunk 3 above)
Comment (xbill82) [reply src/components/Common/CrudlDocument.vue]: Indeed.

File: src/components/Security/Profiles/CommonList.vue (hunk 4 above)
Comment (xbill82) [reply src/components/Security/Profiles/CommonList.vue]: See above.

File: src/components/Security/Profiles/CommonList.vue (hunk 5 above)
Comment (xbill82) [reply src/components/Security/Profiles/CommonList.vue]: It definitely should be, but doesn't seem to be used anywhere. Again, this is not my code, I just formatted it.

File: src/components/Common/CommonList.vue (hunk 6 above)
Comment (xbill82) [reply src/components/Common/CommonList.vue]: good point!

**Pull request merged**